sdk/tests/conformance2/renderbuffers/framebuffer-test.html

```diff
@@ -191,6 +198,131 @@
+    gl.framebufferTexture2D(targetA, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, null, 0);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "detaching a texture from read/draw framebuffer should succeed.");
+  }
+
+  var readFBWithTexture = gl.createFramebuffer();
+  var drawFBWithTexture = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.READ_FRAMEBUFFER, readFBWithTexture);
+  gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, drawFBWithTexture);
+  attachAndDetachTexture(gl.READ_FRAMEBUFFER, gl.DRAW_FRAMEBUFFER);
+  attachAndDetachTexture(gl.DRAW_FRAMEBUFFER, gl.READ_FRAMEBUFFER);
+
+  // attach different textures as color attachment to read and draw framebuffer respectively, then detach these attachments.
+  var fbtex1 = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, fbtex1);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RG8, canvas.width, canvas.height, 0, gl.RG, gl.UNSIGNED_BYTE, null);
+  gl.framebufferTexture2D(gl.DRAW_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, fbtex1, 0);
+  gl.framebufferTexture2D(gl.READ_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, fbtex, 0);
+  shouldBeNonZero('gl.getFramebufferAttachmentParameter(gl.READ_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.FRAMEBUFFER_ATTACHMENT_BLUE_SIZE)');
+  shouldBe('gl.getFramebufferAttachmentParameter(gl.DRAW_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.FRAMEBUFFER_ATTACHMENT_BLUE_SIZE)', '0');
+
+  gl.framebufferTexture2D(gl.READ_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, null, 0);
+  gl.getFramebufferAttachmentParameter(gl.READ_FRAMEBUFFER,
+                                       gl.COLOR_ATTACHMENT0,
+                                       gl.FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION,
+            "calling getFramebufferAttachmentParameter with pname = GL_FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING on read framebuffer without any attachment should generate INVALID_OPERATION.");
+  gl.getFramebufferAttachmentParameter(gl.DRAW_FRAMEBUFFER,
+                                       gl.COLOR_ATTACHMENT0,
+                                       gl.FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "calling getFramebufferAttachmentParameter with pname = GL_FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING on draw framebuffer should succeed.");
+  gl.framebufferTexture2D(gl.DRAW_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, null, 0);
+  gl.getFramebufferAttachmentParameter(gl.DRAW_FRAMEBUFFER,
+                                       gl.COLOR_ATTACHMENT0,
+                                       gl.FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION,
+            "calling getFramebufferAttachmentParameter with pname = GL_FRAMEBUFFER_ATTACHMENT_COLOR_ENCODING on draw framebuffer without any attachment should generate INVALID_OPERATION.");
```

Comment:
These tests are redundant -- they're already done in attachAndDetachTexture. Please remove them.